`make all` results in error `scope variable ``writer`` assigned to non-scope parameter ``writer`` calling util.writer.finishWriterToSafeCStr`

**Describe problem**
Can't build the crow website. During build `scope variable ``writer`` assigned to non-scope parameter ``writer`` calling util.writer.finishWriterToSafeCStr` error occurs in `src/interpret/runBytecode.d(319)` file

**Expected behavior**
Website builds and serves

**Actual behavior**
Website fails to build

**Steps to reproduce**
1) Clone the repo
2) Install dependencies as per `readme.md` (`ldc`,`dmd`,`hg`)
3) run `make all`

**Additional info**
System: Arch linux
Kernel: 5.16.15

<details>
<summary>Build log</summary>

```bash
rm -rf bin site temp
mkdir -p bin/d-imports
date --iso-8601 --utc > bin/d-imports/date.txt
mkdir -p bin/d-imports
git rev-parse --short HEAD > bin/d-imports/commit-hash.txt
hg clone https://dyncall.org/pub/dyncall/dyncall/
destination directory: dyncall
requesting all changes
adding changesets
adding manifests
adding file changes
added 468 changesets with 2242 changes to 724 files                                                                                                                                            
new changesets 3e629dc19168:b47168dacba6
updating to branch default
582 files updated, 0 files merged, 0 files removed, 0 files unresolved
cd dyncall && ./configure
Configuration written to Makefile.config
cd dyncall && make
make[1]: Entering directory '/home/boris/Desktop/crow/dyncall'
cd dyncall && make all
make[2]: Entering directory '/home/boris/Desktop/crow/dyncall/dyncall'
cc -fPIC   -c -o dyncall_vector.o dyncall_vector.c
cc -fPIC   -c -o dyncall_api.o dyncall_api.c
cc -fPIC   -c -o dyncall_callvm.o dyncall_callvm.c
cc -fPIC   -c -o dyncall_callvm_base.o dyncall_callvm_base.c
cc    -c -o dyncall_call.o dyncall_call.S
cc -fPIC   -c -o dyncall_callf.o dyncall_callf.c
cc -fPIC   -c -o dyncall_struct.o dyncall_struct.c
ar rv libdyncall_s.a dyncall_vector.o dyncall_api.o dyncall_callvm.o dyncall_callvm_base.o dyncall_call.o dyncall_callf.o dyncall_struct.o
ar: creating libdyncall_s.a
a - dyncall_vector.o
a - dyncall_api.o
a - dyncall_callvm.o
a - dyncall_callvm_base.o
a - dyncall_call.o
a - dyncall_callf.o
a - dyncall_struct.o
make[2]: Leaving directory '/home/boris/Desktop/crow/dyncall/dyncall'
cd dyncallback && make all
make[2]: Entering directory '/home/boris/Desktop/crow/dyncall/dyncallback'
cc -fPIC -I/home/boris/Desktop/crow/dyncall/./dyncallback/../dyncall    -c -o dyncall_alloc_wx.o dyncall_alloc_wx.c
cc -fPIC -I/home/boris/Desktop/crow/dyncall/./dyncallback/../dyncall    -c -o dyncall_args.o dyncall_args.c
cc -fPIC -I/home/boris/Desktop/crow/dyncall/./dyncallback/../dyncall    -c -o dyncall_callback.o dyncall_callback.c
cc    -c -o dyncall_callback_arch.o dyncall_callback_arch.S
cc -fPIC -I/home/boris/Desktop/crow/dyncall/./dyncallback/../dyncall    -c -o dyncall_thunk.o dyncall_thunk.c
ar rv libdyncallback_s.a dyncall_alloc_wx.o dyncall_args.o dyncall_callback.o dyncall_callback_arch.o dyncall_thunk.o
ar: creating libdyncallback_s.a
a - dyncall_alloc_wx.o
a - dyncall_args.o
a - dyncall_callback.o
a - dyncall_callback_arch.o
a - dyncall_thunk.o
make[2]: Leaving directory '/home/boris/Desktop/crow/dyncall/dyncallback'
cd dynload && make all
make[2]: Entering directory '/home/boris/Desktop/crow/dyncall/dynload'
cc -fPIC   -c -o dynload.o dynload.c
cc -fPIC   -c -o dynload_syms.o dynload_syms.c
ar rv libdynload_s.a dynload.o dynload_syms.o
ar: creating libdynload_s.a
a - dynload.o
a - dynload_syms.o
make[2]: Leaving directory '/home/boris/Desktop/crow/dyncall/dynload'
make[1]: Leaving directory '/home/boris/Desktop/crow/dyncall'
dmd -ofbin/crow-debug -betterC -preview=dip25 -preview=dip1000 -J=bin/d-imports -check=on -boundscheck=on -debug -g -version=Debug -version=Test src/app.d src/concretize/*.d src/frontend/*.d src/frontend/*/*.d src/interpret/*.d src/lib/*.d src/lower/*.d src/model/*.d src/util/*.d src/util/*/*.d src/backend/*.d src/document/*.d src/test/*.d -L=-ldyncall_s -L=-L./dyncall/dyncall -L=-lgccjit
src/interpret/runBytecode.d(319): Error: scope variable `writer` assigned to non-scope parameter `writer` calling util.writer.finishWriterToSafeCStr
src/interpret/runBytecode.d(332): Error: scope variable `writer` assigned to non-scope parameter `writer` calling util.writer.finishWriterToSafeCStr
make: *** [Makefile:78: bin/crow-debug] Error 1
```
</details>

If comment out lines that result in error, another error occurs: 
```bash
/usr/bin/ld: cannot find -lgccjit
collect2: error: ld returned 1 exit status
Error: linker exited with status 1
```
After `lgccjit` installation another error occurs: 
```bash
cannot find program `wasm-ld`
```
After `wasm-ld` installation another error occurs `./readme.md line 1 is 164 columns long, should be <= 120`
After installing missing dependencies and uncommenting commented lines the error `scope variable ``writer`` assigned to non-scope parameter ``writer`` calling` is still present

After removing first line of `readme.md` and commenting out `src/interpret/runBytecode.d(332): ` it finally starts.

--- a/readme.md
+++ b/readme.md
@@ -1,4 +1,5 @@
-[![Gitter](https://badges.gitter.im/crow-lang-org/community.svg)](https://gitter.im/crow-lang-org/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
+[![Gitter](https://badges.gitter.im/crow-lang-org/community.svg)]
+(https://gitter.im/crow-lang-org/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
 
 # Crow
 
@@ -11,10 +12,12 @@ For information about the langauge itself, visit the [website](http://crow-lang.
 
 Install these tools (potentially from your operating system's package manager):
 
-* [`git`](https://git-scm.com/) (used to get this repository)
-* [`ldc`](https://github.com/ldc-developers/ldc#installation) (used to compile `bin/crow`).
 * [`dmd`](https://dlang.org/download.html#dmd) (used to compile `bin/crow-debug` due to much faster compiles)
-* [`hg`](http://mercurial-scm.org) (used to clone the dyncall library)
+* [`git`](https://git-scm.com/) (used to get this repository)
+* [`hg`](https://mercurial-scm.org) (used to clone the `dyncall` library)
+* [`ldc`](https://github.com/ldc-developers/ldc#installation) (used to compile `bin/crow`).
+	- `wasm-ld` may need to be installed separately (used to compile `bin/crow.wasm`)
+* [`libgccjit`](https://gcc.gnu.org/onlinedocs/jit) (`bin/crow` links to this)
 
 Then run:
 

--- a/src/util/writer.d
+++ b/src/util/writer.d
@@ -24,7 +24,7 @@ immutable(string) finishWriter(scope ref Writer writer) {
 	return finishWriter(writer).ptr;
 }
 
-@trusted immutable(SafeCStr) finishWriterToSafeCStr(ref Writer writer) {
+@trusted immutable(SafeCStr) finishWriterToSafeCStr(scope ref Writer writer) {
 	return immutable SafeCStr(finishWriterToCStr(writer));
 }
 

